wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -459,6 +486,49 @@ private void validateInputSchema(@Nullable Schema inputSchema, FailureCollector
     }
   }
 
+  private void validatePrecondition(String precondition, Boolean isConditionSQL, FailureCollector collector) {
+    String field = Config.NAME_PRECONDITION;
+    String language = "Precondition (JEXL)";
+
+    if (isConditionSQL == true) {
+      field = Config.NAME_PRECONDITION_SQL;
+      language = "Precondition (SQL)";
+    }
+    if (precondition == null) {
+      collector.addFailure(String.format("%s must be present.", language),
+                           null)
+        .withConfigProperty(field);
+    }
+  }
+
+  private void validateDirectives(FailureCollector collector) {
+    if (config.directives != null) {
+      collector.addFailure("Directives are not supported for precondition of type SQL", null)
+        .withConfigProperty(Config.NAME_DIRECTIVES);
+    }
+
+    if (config.udds != null) {
+      collector.addFailure("UDDs are not supported for precondition of type SQL", null)
+        .withConfigProperty(Config.NAME_UDD);
+    }
+  }
+
+  private boolean checkPreconditionNotEmpty(Boolean isConditionSQL) {
+    if (isConditionSQL != false) {
+      if (!config.preconditionLanguage.equals("sql") && config.precondition != null
```

Comment:
nit: generally we should keep constants first to avoid NPE i.e. `"sql".equals(config.preconditionLanguage))` and replace other two checks with `Strings.isNullOrEmpty(config.precondition)`.